ch-signup validation

nwashangai opened the pull request:
#### What does this PR do?
it validates signup input
#### Description of Task to be completed?
* update validation
* write tests
#### How should this be manually tested?
* Not applicable
#### Any background context you want to provide?
* None
#### What are the relevant pivotal tracker stories?
#160702881
#### Screenshots (if appropriate)
* None
#### Questions:
* None

nwashangai committed:
ch-signup validation
    - validate user signup inputs
    [Finishes #160702881]

nwashangai committed:
ft- user login
    - create login endpoint
    - validate inputs
    - write test for login edpoint
    [Finishes #160702910]

nwashangai committed:
ft: place order
    - create endpoint for placing order
    - vaidate ndpoint
    - write test for place order endpoint
    [Finishes #160702928]

dogoyaro commented:
Please title this PR according to the standards we agreed upon.

dogoyaro commented on .travis.yml:
```diff
@@ -13,11 +13,15 @@ addons:
   - ./node_modules/mocha/bin/mocha --recursive './test/' --compilers js:babel-core/register --exit
+  - psql -c 'drop database food;' -U postgres
 after_success:
+  - psql -c 'create database food;' -U postgres
```
Are you trying to drop this database? Not sure why you're creating after success. Please explain what this does.

nwashangai replied:
in Travis after running the test, for me to run the cover which has to run another test, I have to recreate the schemas in order not to  duplicate the entry in the tables thereby raising error

dogoyaro commented on src/middlewares/validator.js:
```diff
@@ -1,8 +1,11 @@
     let isValid = 1;
-    if (request.body.foodItems) {
+    if (request.method === 'POST' && (request.originalUrl === '/api/v1/orders/' || request.originalUrl === '/api/v1/orders')) {
+        const count = (request.body.foodItems) ? (Array.isArray(request.body.foodItems)) ? request.body.foodItems.length : 0 : 0;
```
Aren't if statements more readable than nested ternary operators?

dogoyaro commented on src/middlewares/validator.js:
```diff
@@ -1,8 +1,11 @@
-    if (request.body.foodItems) {
+    if (request.method === 'POST' && (request.originalUrl === '/api/v1/orders/' || request.originalUrl === '/api/v1/orders')) {
+        const count = (request.body.foodItems) ? (Array.isArray(request.body.foodItems)) ? request.body.foodItems.length : 0 : 0;
+        if (!request.body.foodItems === undefined || !count < 1) {
```
I think you only need to check the count here, as the check above should have handled that check. Sync with me if you have questions about this.

dogoyaro commented on src/middlewares/validator.js:
```diff
@@ -12,11 +15,31 @@ export default (request, response, next) => {
             return (response.status(400).json({ status: 'error', message: isValid }));
         }
+    } else if (request.method === 'POST' && (request.originalUrl === '/api/v1/auth/signup' || request.originalUrl === '/api/v1/auth/signup/')) {
+      const regex = /^[a-zA-Z0-9._-]+@[a-zA-Z0-9.-]+\.[a-zA-Z]{2,4}$/;
```
Do you think it might be advantageous to move the logic that validates the email to some kind of helper method, so you can re-use for all the places where you have to validate emails? Perhaps you could even do that for other fields.

dogoyaro commented on src/middlewares/validator.js:
```diff
@@ -12,11 +15,31 @@ export default (request, response, next) => {
+      const regex = /^[a-zA-Z0-9._-]+@[a-zA-Z0-9.-]+\.[a-zA-Z]{2,4}$/;
+      isValid = (!regex.test(String(request.body.email)) || !request.body.email) ?
+          'Invalid email id': (!/^[A-Za-z]+$/.test(String(request.body.name)) || !request.body.name) ?
+          'Invalid name': (!request.body.phone || !/^\d{11}$/.test(request.body.phone)) ?
```
Same as above.

dogoyaro commented on test/index.js:
```diff
@@ -20,4 +21,13 @@ describe('Fast-Food-Fast', () => {
     });
   });
+  describe('Migration test', () => {
+    it('it should Successfully migrate database schemas', (done) => {
```
This test has no assertions. Please fix.

nwashangai committed:
ft: get order history for a particular user
    - crreate endpoint to get order history of user
    - write test for the endpoint
    [Finishes #160702946]

nwashangai committed:
Merge pull request #36 from nwashangai/ft-user-order-history-2197024

nwashangai committed:
ft: Admin can add food
    - create endpoint to add food
    - validate user inputs
    - write test cases for this endpoint
    [Finishes #160703460]

nwashangai committed:
ft: get food menu
    - create endpoint to get food menu
    - write test for this endpoint
    [Finishes #160703275]

nwashangai committed:
ft: update order status
    - create endpiont to update status
    - write test cases for this endpoint
    [Finishes #160703227]

nwashangai committed:
ft: admin gets specific order
    - create endpoint to get specific order
    - write test cases for this endpoint
    [Finishes #160703172]

nwashangai committed:
ft: admin gets all orders
    - create endpoint to get all orders
    - write test cases for this endpoint
    [Finishes #160703159]

nwashangai committed:
ft: adds update menu item functionalities
    - create endpoint to update menu item
    - write test cases for this endpoint
    [Finishes #160703481]

nwashangai committed:
Merge pull request #42 from nwashangai/ft-update-menu-item-2197024

#160703481 adds update menu item functionalities

nwashangai committed:
Merge pull request #41 from nwashangai/ft-get-all-orders-2197024

#160703159 admin gets all orders

nwashangai committed:
Merge pull request #40 from nwashangai/ft-get-specific-2197024

#160703172 admin gets specific order

nwashangai committed:
Merge pull request #39 from nwashangai/ft-update-status-2197024

nwashangai committed:
Merge pull request #38 from nwashangai/ft-get-menu-2197024

#160703275 get food menu

nwashangai committed:
Merge pull request #37 from nwashangai/ft-add-food-2197024

nwashangai committed:
Merge pull request #35 from nwashangai/ft-place-order-2197024

nwashangai committed:
Merge pull request #34 from nwashangai/ft-login-2197024
